### ISSUE TITLE
Problem setting application stack for linux web app and java 8

### ISSUE THREAD

**Issue by PbALpi7xEX** [hashicorp/terraform-provider-azurerm#16200]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.17

### AzureRM Provider Version

3.0.2

### Affected Resource(s)/Data Source(s)

azurerm_linux_web_app

### Terraform Configuration Files

```hcl
application_stack {
      # pin java version in production to avoid unexpected outages when JVM auto updates
      # need to update as versions come out if pinning however
      java_version = "1.8" 
      java_server  = "JAVA"
      java_server_version = terraform.workspace == "dev" ? "1.8.0_312" : "1.8" 
}
```

### Debug Output/Panic Output

```shell
n/a
```

### Expected Behaviour

Expecting java stack settings to be set to 
Stack: Java 
Major Version: Java 8
Java web server: Java SE (embedded)
Java web server version: Java SE 8u312

### Actual Behaviour

Stack settings empty.

State File shows the following:
 "application_stack": [
                  {
                    "docker_image": "",
                    "docker_image_tag": "",
                    "dotnet_version": "",
                    "java_server": "JAVA",
                    "java_server_version": "1.8.0_312",
                    "java_version": "1.8",
                    "node_version": "",
                    "php_version": "",
                    "python_version": "",
                    "ruby_version": ""
                  }no
"linux_fx_version": "JAVA|1.8.0_312-1.8"

**Comment by joshft91** [comment]
Seeing the same here. Using `azurerm_linux_function_app`, the following application stack(s) result in Terraform seeing them as a new change every time `terraform plan` is run.

```hcl
application_stack {
  node_version = "14"
}

application_stack {
  python_version = "3.9"
}
```

**Comment by PbALpi7xEX** [comment]
so using this syntax is working for me now:

java_version        = terraform.workspace == "prod" ? "8u312" : "jre8"
java_server         = "JAVA"
java_server_version = terraform.workspace == "prod" ? "" : "8"

Maybe just me, but I found the docs a bit misleading around the application_stack acceptable values... ?

**Comment by zemuldo** [comment]
Same issue here with 
```sh
application_stack {
  python_version = "3.9"
}
```

**Comment by preetishmadalia** [comment]
Same issue with 
`application_stack         =  [{
                        java_version        = "JAVA|17-java17"
                      }]`

**Comment by xiaxyi** [comment]
#16375

**Comment by p0onage** [comment]
Same issue here on windows app the stack settings are shown in the plan but missing on the actual app.

`  site_config {
    always_on = true
    application_stack {
      current_stack       = "dotnet"
      dotnet_version      = "v6.0"
    }`

**Comment by shankaranand2013** [comment]
Same issue here but the resource is "azurerm_logic_app_standard" .

I am using an app service of kind = "Linux" and reserved = "True"

The site config block has the following line for the stack settings,

Site_config = {
      linux_fx_version                 = "NODE|12-lts"
     }

But I am seeing the stack as ".net" and .net version , Instead I should be seeing Node.js .
<img width="471" alt="logicapp_standard" src="https://user-images.githubusercontent.com/71544036/184936029-ddd04fd2-c0d4-4d68-bb7b-58d487323f19.PNG">

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. It looks like this has been resolved as of #19685 which supersedes #18155. As such, I am going to mark this issue as closed.
